### ISSUE TITLE
rails5 // invalid uri scheme

### ISSUE THREAD

**Issue by bbnnt** [redis-store/redis-rails#42]
sorry if it does not fits, but feels like something buggy; never had those problems using redis this way

with the following Gemfile part (taken from https://github.com/redis-store/redis-rails/issues/30#issuecomment-209810069) :

```
...
gem 'redis'
gem 'redis-namespace'
gem 'redis-rack', github: 'redis-store/redis-rack', branch: 'master'
gem 'redis-actionpack', github: 'redis-store/redis-actionpack', branch: 'master'
gem 'redis-rails', github: 'redis-store/redis-rails', branch: 'master'
...
```

I keep being hit with the following error while trying to start the app :

`/...path_to_rvm.../.rvm/gems/ruby-2.3.1/gems/redis-3.3.0/lib/redis/client.rb:419:in _parse_options: invalid uri scheme '' (ArgumentError)`

I have this :

```
config.cache_store = :redis_store, {  scheme: :redis,
                                          host: "localhost",
                                          port: 6379,
                                          db: 0,
                                          namespace: "cache",
                                          expires_in: 90.minutes }

```

I'm not that of a gem developper, but opening the file and tracing what happened in this part, the uri only contains the host, so well it can"t find the scheme, and other parts… weirdly (or not), the uri trace returns 0.0.0.0 instead of localhost

Assuming the README (https://github.com/redis-store/redis-rails#usage) is still relevant. In any cases, am pretty stuck; hope it seems obvious to some

**Comment by tubbo** [comment]
Try not setting the `:scheme`, as I believe that's inferred.

**Comment by bbnnt** [comment]
thx for your reply

I started with the basic one : `config.cache_store = :redis_store, "redis://localhost:6379/0/cache", { expires_in: 90.minutes }`

it fails here : https://github.com/redis/redis-rb/blob/master/lib/redis/client.rb#L404

anything done, at this position the `url` variable you see there is only the host, what is sent in the conf seems stripped at a higher level (you put `redis://localhost:6379/0/cache` in the rails conf it logs only `localhost`, you put `redis://0.0.0.0:6379/0/cache` it logs `0.0.0.0`, etc…) 

The `URI(url)` won't output much more then, `uri = URI(url); uri.scheme` then breaks

**Comment by tubbo** [comment]
That's slightly different from what you posted before. Try not setting the scheme but keeping all the other attributes.

**Comment by tubbo** [comment]
Check out https://github.com/redis-store/redis-rails/issues/17#issuecomment-88409459, noticed you were setting `expires_at` which doesn't really work. I also updated the README.

**Comment by bbnnt** [comment]
Hello, thanks for your reply

I did not configured both url and attributes, I just tried both

I'm sorry to insist, still on the same link https://github.com/redis/redis-rb/blob/master/lib/redis/client.rb#L404, it expects an object but it fails as a string is passed (and in my case the string is only the domain — localhost)

I copy pasted your README, `bundle update`'d the gems, Gemfile.lock show 4.0.0.

Same error.

Used to work very well.
